packages/hardhat-core/src/internal/solidity/compiler/index.ts

```diff
@@ -68,14 +71,39 @@ export class Compiler implements ICompiler {
 }
 
 export class NativeCompiler implements ICompiler {
-  constructor(private _pathToSolc: string) {}
+  constructor(private _pathToSolc: string, private _solcVersion?: string) {}
 
   public async compile(input: CompilerInput) {
+    const args = ["--standard-json"];
+
+    // Logic to make sure that solc default import callback is not being used.
+    // If solcVersion is not defined or <= 0.6.8, do not add extra args.
+    if (this._solcVersion !== undefined) {
+      if (semver.parse(this._solcVersion) === null) {
+        throw new HardhatError(ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE, {
```

Comment:
This error belongs to a different layer, because it's about _tasks_ arguments. While right now the `_solcVersion` value comes directly from a task argument, this class shouldn't be aware of that. I see two possible solutions:

1. Replace this with an assertion error (perhaps using `assertHardhatInvariant`, even if this is not exactly an invariant).
2. Move this check to `TASK_COMPILE_SOLIDITY_RUN_SOLC` and just assume that the version is valid here.

I don't have a preference.